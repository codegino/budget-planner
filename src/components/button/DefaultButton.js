import React from 'react';
import { Text, View, StyleSheet, TouchableNativeFeedback, Platform, TouchableOpacity } from 'react-native';
import colors from '../../constants/colors';

const styles = StyleSheet.create({
  container: {
    backgroundColor: colors.primaryDark,
    height: 50,
    width: '100%',
    borderRadius: 20,
    borderWidth: 1,
    borderColor: colors.primaryLight,
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    color: 'white',
  },
});

type Props = {
  style: Object,
  title: string,
};


const buttonWithBackground = (props: Props) => {
  const content = (
    <View style={[styles.container, props.style]}>
      <Text style={styles.title}>{props.title}</Text>
    </View>
  );

  if (Platform.OS === 'android') {
    return (
      <TouchableNativeFeedback {...props}>
        {content}
      </TouchableNativeFeedback>
    );
  }
  return <TouchableOpacity {...props}>{content}</TouchableOpacity>;
};

export default buttonWithBackground;
