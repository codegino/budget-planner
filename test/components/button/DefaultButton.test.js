import React from 'react';
import DefaultButton from 'components/button/DefaultButton';
import renderer from 'react-test-renderer';

describe('Default Button', () => {
  it('will have have a text of hello', () => {
    const button = <DefaultButton title="Hello" />;
    expect(renderer.create(button).toJSON()).toMatchSnapshot();
  });

  it('will have background color of red', () => {
    const button = <DefaultButton title="Hello" style={{ backgroundColor: 'red' }} />;
    expect(renderer.create(button).toJSON()).toMatchSnapshot();
  });
});
