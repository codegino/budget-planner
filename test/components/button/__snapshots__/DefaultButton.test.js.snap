// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Button will have background color of red 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "red",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#193e2c",
          "borderColor": "#5fbd98",
          "borderRadius": 20,
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "width": "100%",
        },
        Object {
          "backgroundColor": "red",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      Hello
    </Text>
  </View>
</View>
`;

exports[`Default Button will have have a text of hello 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#193e2c",
          "borderColor": "#5fbd98",
          "borderRadius": 20,
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      Hello
    </Text>
  </View>
</View>
`;
